### DISCUSSION TITLE
Docker image with Ollama installed very slow

### DISCUSSION THREAD

**Discussion by PietFourie** [open-webui/open-webui#2309] [General]
Hi
My system has a NVDIA 1070 8 GB GPU card running linux mint

I used the latest Docker images as installed on the 15th May

1) I have installed the Docker image with Ollama installed and it runs very slow.
2) I then installed the Docker image with Ollama and Open WebUI seperately and it runs the fastest.
3) I then ran the same prompt directly on the Ollama terminal and it had similar speed. 

The prompt was :
Write me a python code for the game centipede
Model LLama 3:lastest

1) The code generated by Ollama directly and with a seperate Ollama/Open WebUI docker image was the same and used the pygame library. I did not test the code but is used clasees, and more advanced code
2) The code generated by the combined Olllama/Webui docker image create a very small code using no libraries and I doubt it would have run.

The error is that the combined Ollama/OpenWebUI docker miage creates carbage and is a lot slower.

Error with ModelFilles
Also I coulld not import the modelfiles from the website. It just does nothing. Also when you import the model it looks for ".json" files but the downloaded modelfile have a ".txt"  extension. I would presume the error is significant. All model files are on the recommended home page of the website and should be compatible.

**Comment by PietFourie** [comment]
Here is the code generated
[Ollama direct Inference Lama3.txt](https://github.com/open-webui/open-webui/files/15331931/Ollama.direct.Inference.Lama3.txt)
[Open WEBUI and Ollama Seperated Docker images.txt](https://github.com/open-webui/open-webui/files/15331932/Open.WEBUI.and.Ollama.Seperated.Docker.images.txt)
[OpenWebUI with Ollama InStalled Inference Lama3.txt](https://github.com/open-webui/open-webui/files/15331933/OpenWebUI.with.Ollama.InStalled.Inference.Lama3.txt)

**Comment by PietFourie** [comment]
GPU memory used in the combined version was lower than in the separate Ollama docker images. It also used CPU cores.

**Comment by Kaszebe** [comment]
I have the same problem.

I have 4x3090, 1x4080, and 1x4090. I installed OpenWeb UI docker and Ollama (no mods/etc). It's literally slower than a snail when performing inference with a 35B model. I have 136GB of VRAM...this model (Wizard coder) should be flying instead of crawling.

**Comment by NaturallyAsh** [comment]
Has this been addressed because I'm experiencing something similar? Docker Ollama with Mistral model takes a minute, usually longer, to execute in Open WebUI, but the same model executes immediately and fast in LM Studio and terminal.
